core/runtime/jsruntime.rs

```diff
@@ -1410,45 +1307,60 @@ impl JsRuntime {
 
   pub fn poll_value(
     &mut self,
+    cx: &mut Context,
     global: &v8::Global<v8::Value>,
+  ) -> Poll<Result<v8::Global<v8::Value>, Error>> {
+    Self::with_runtime_scope(
+      self.v8_isolate_ptr(),
+      self.inner.main_realm.context_ptr(),
+      |scope| self.poll_value_inner(cx, scope, global),
+    )
+  }
+
+  fn poll_value_inner(
+    &mut self,
     cx: &mut Context,
+    scope: &mut v8::HandleScope,
+    global: &v8::Global<v8::Value>,
   ) -> Poll<Result<v8::Global<v8::Value>, Error>> {
+    let promise = v8::Local::new(scope, global);
+
+    // TODO(mmastrac): this would be better if we just type-tested at the top level
+    let Ok(promise) = v8::Local::<v8::Promise>::try_from(promise) else {
+      return Poll::Ready(Ok(global.clone()));
+    };
```

Comment:
I'd be in favor of that, makes sense to panic here if it's not a `v8::Promise`.

Reply:
For sure... I want to leave it as a TODO because we don't have a way to cast/type test `v8::Global<...>` in rusty_v8 like we do with `v8::Local<...>` uet. I suspect it's a bit of a hole in our rusty_v8 API.